src/trade.js

```diff
@@ -152,7 +161,8 @@ class Trade {
     var reservation = quote.delegate.reserveReceiveAddress();
 
     var processTrade = function (res) {
-      var trade = new quote._TradeClass(res, quote.api, quote.delegate);
+      var trade = new quote._TradeClass(null, quote.api, quote.delegate);
+      trade.setFromAPI(res);
```

Comment:
```js
var trade = new quote._TradeClass(null, quote.api, quote.delegate);
trade.setFromAPI(res);
```
When you instantiate `Trade` you need to pass it the info from the KV Store, not from the API. This confused me a few times as well during the refactor. The KV Store contains information that the API doesn't, e.g. the `account_index` and `receive_index`. 

This particular line is inside `buy()`; when creating a new order there is no KV store entry yet, which is why I pass in `null`.